pkg/stacks/unpack.go

```diff
@@ -576,5 +595,30 @@ func (sp *StackPackage) applyUISpecAnnotations(crdPath string, crd *apiextension
 			}
 		}
 	}
+}
+
+// isMetadataApplicableToCRD determines if the given metadata file path is applicable to the given CRD.
+func isMetadataApplicableToCRD(crdPath, metadataPath string, globalFileNames []string, crdKind string) bool {
+	// compare the directory of the given metadata file path to the CRDs path
+	metadataDir := filepath.Dir(metadataPath)
```

Comment:
Nitpick: I am thinking we may want to make a note for future maintainers that this implementation assumes that there is at least one level of directories above the path, or that the paths are all absolute (because `filepath.Dir` [can return](https://golang.org/pkg/path/filepath/#Dir) `"."` otherwise)

Reply:
I think this comment actually applies at a higher level and not just this function.  I think it applies to the entire unpack processing.  The way the Stack Manager invokes this unpacking will always result in a directory structure with a root, so we shouldn't run into any problems in practical usage, but it's probably worth adding a top level comment.

Reply:
Pushed comments in all applicable places that these paths should be absolute (including the applicable main.go argument).